aws-android-sdk-pinpoint/src/main/java/com/amazonaws/mobileconnectors/pinpoint/targeting/notification/NotificationClient.java

```diff
@@ -284,16 +284,27 @@ public String getChannelType() {
      * local broadcast is sent.
      *
      * @param notificationDetails the notification message received by the device's messaging service
-     * @return {@link NotificationClient.CampaignPushResult}.
+     * @return {@link PinpointPushResult}.
+     *
+     */
+    public PinpointPushResult handlePushNotification(final NotificationDetails notificationDetails) {
+        return notificationClientBase.handlePushNotification(notificationDetails);
```

Comment:
Ughh. This isn't your bad, I'm just commenting on the existing architecture here of this legacy code base .... when you have a `Foo`, and a `FooBase`, you'd expect `Foo extends FooBase`, right? But no. Here, `Foo` "has a" `FooBase` ah man.

Reply:
Yeah...that was very confusing. I thought about going through and taking a stab at making this a little cleaner, but the scope started to get a little bit larger. On top of that, they created classes that then inherit from `Foo` for each of the push notification services 🤦‍♂